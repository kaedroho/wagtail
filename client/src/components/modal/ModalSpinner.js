import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  isActive: PropTypes.bool,
  children: PropTypes.node,
};

const defaultProps = {
  isActive: false,
  children: null,
};

const ModalSpinner = ({ isActive, children }) =>
  <div className={`loading-mask${isActive ? ' loading' : ''}`}>
    {children}
  </div>;

ModalSpinner.propTypes = propTypes;
ModalSpinner.defaultProps = defaultProps;

export default ModalSpinner;
