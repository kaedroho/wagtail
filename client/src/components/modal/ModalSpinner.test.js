import React from 'react';
import { shallow } from 'enzyme';
import ModalSpinner from './ModalSpinner';

describe('ModalSpinner', () => {
  it('renders', () => {
    expect(shallow((
      <ModalSpinner>
        Test
      </ModalSpinner>
    ))).toMatchSnapshot();
  });

  it('#isActive', () => {
    expect(shallow((
      <ModalSpinner isActive={true}>
        Test
      </ModalSpinner>
    ))).toMatchSnapshot();
  });
});
