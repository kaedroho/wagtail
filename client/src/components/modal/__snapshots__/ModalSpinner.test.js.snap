// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalSpinner #isActive 1`] = `
<div
  className="loading-mask loading"
>
  Test
</div>
`;

exports[`ModalSpinner renders 1`] = `
<div
  className="loading-mask"
>
  Test
</div>
`;
